File: frontend/src/types/cliente.ts
// Tipagens do módulo de CRM de Clientes/Leads

export interface Cliente {
  id?: string;
  conta_id?: string;
  nome: string;
  cpf?: string;
  data_nascimento?: string; // YYYY-MM-DD
  estado_civil?: string;
  email?: string;
  telefone?: string;
  status_funil: string;
  origem?: string;
  interesse_tipo?: string;
  notas?: string;
  tags?: string[];
  preferencias?: ClientePreferencias;
  financeiro?: ClienteFinanceiro;
  origem_utm?: ClienteOrigemUTM;
  corretor_id?: string;
  temperatura?: string; // Quente, Morno, Frio
  proxima_acao?: string;
  criado_em?: string;
  atualizado_em?: string;
}

export interface ClientePreferencias {
  tipo_imovel?: string[];
  finalidade?: string;
  orcamento_min?: number | null;
  orcamento_max?: number | null;
  bairros?: string[];
  quartos_min?: number | null;
}

export interface ClienteFinanceiro {
  renda_mensal?: number | null;
  precisa_financiamento?: string; // Sim, Nao, JaAprovado
  possui_fgts?: boolean;
  forma_pagamento?: string; // AVista, Financiamento, Consorcio, Parcelado
}

export interface ClienteOrigemUTM {
  canal?: string;
  campanha?: string;
  imovel_origem_id?: string;
}

export interface Interacao {
  id?: string;
  conta_id?: string;
  cliente_id: string;
  corretor_id?: string;
  tipo: string; // LIGACAO, MENSAGEM, VISITA, PROPOSTA, ANOTACAO
  descricao: string;
  data_hora?: string;
  criado_em?: string;
}

// Etapas do funil de vendas
export const STATUS_FUNIL = [
  { value: 'NOVO',         label: 'Novo Contato'    },
  { value: 'ATENDIMENTO',  label: 'Em Atendimento'  },
  { value: 'VISITA',       label: 'Visita Agendada' },
  { value: 'PROPOSTA',     label: 'Proposta'        },
  { value: 'FECHADO',      label: 'Fechado/Ganho'   },
  { value: 'PERDIDO',      label: 'Perdido'         },
];

// Origens possíveis de um lead
export const ORIGEM_LEAD = [
  { value: 'SITE',       label: 'Site Próprio'       },
  { value: 'PORTAL',     label: 'Portal Imobiliário' },
  { value: 'WHATSAPP',   label: 'WhatsApp'           },
  { value: 'INDICACAO',  label: 'Indicação'          },
  { value: 'SOCIAL',     label: 'Redes Sociais'      },
  { value: 'OUTROS',     label: 'Outros'             },
];

File: backend/internal/core/domain/cliente.go
package domain

import "time"

type Cliente struct {
	ID             string               `json:"id"`
	ContaID        string               `json:"conta_id"`
	Nome           string               `json:"nome"`
	CPF            *string              `json:"cpf,omitempty"`
	DataNascimento *string              `json:"data_nascimento,omitempty"` // YYYY-MM-DD
	EstadoCivil    *string              `json:"estado_civil,omitempty"`
	Email          *string              `json:"email"`
	Telefone       *string              `json:"telefone"`
	StatusFunil    string               `json:"status_funil"` // NOVO, ATENDIMENTO, VISITA, PROPOSTA, FECHADO
	Origem         *string              `json:"origem"`
	InteresseTipo  *string              `json:"interesse_tipo"`
	Notas          *string              `json:"notas"`
	Tags           []string             `json:"tags"`
	Preferencias   *ClientePreferencias `json:"preferencias"`
	Financeiro     *ClienteFinanceiro   `json:"financeiro,omitempty"`
	OrigemUTM      *ClienteOrigemUTM    `json:"origem_utm,omitempty"`
	CorretorID     *string              `json:"corretor_id,omitempty"`
	Temperatura    *string              `json:"temperatura,omitempty"` // Quente, Morno, Frio
	ProximaAcao    *time.Time           `json:"proxima_acao,omitempty"`
	CriadoEm       time.Time            `json:"criado_em"`
	AtualizadoEm   time.Time            `json:"atualizado_em"`
}

type ClienteFinanceiro struct {
	RendaMensal          *float64 `json:"renda_mensal,omitempty"`
	PrecisaFinanciamento *string  `json:"precisa_financiamento,omitempty"` // Sim, Nao, JaAprovado
	PossuiFGTS           *bool    `json:"possui_fgts,omitempty"`
	FormaPagamento       *string  `json:"forma_pagamento,omitempty"` // AVista, Financiamento, Consorcio, Parcelado
}

type ClienteOrigemUTM struct {
	Canal          string `json:"canal,omitempty"`
	Campanha       string `json:"campanha,omitempty"`
	ImovelOrigemID string `json:"imovel_origem_id,omitempty"`
}

type ClientePreferencias struct {
	TipoImovel   []string `json:"tipo_imovel,omitempty"` // Apartamento, Casa, Lote, Comercial, Rural
	Finalidade   string   `json:"finalidade,omitempty"`  // Compra, Locacao
	OrcamentoMin *float64 `json:"orcamento_min,omitempty"`
	OrcamentoMax *float64 `json:"orcamento_max,omitempty"`
	Bairros      []string `json:"bairros,omitempty"`
	QuartosMin   *int     `json:"quartos_min,omitempty"`
}
